### PR TITLE
Standardize and improve test suite

### PR DESCRIPTION
**Pull request by acoven**
Add comprehensive Jest test suites for core classes.

These new test suites improve test coverage and align with the requested `test_markdownify.js` pattern, providing one thorough automated test suite per class.

---

[Open in Web](https://www.cursor.com/agents?id=bc-5357c266-1ea7-4cf1-83cf-14723f9da093) • [Open in Cursor](https://cursor.com/background-agent?bcId=bc-5357c266-1ea7-4cf1-83cf-14723f9da093)

Learn more about [Background Agents](https://docs.cursor.com/background-agent/web-and-mobile)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive Jest test suites for key classes, including App, Chrome, EventTarget, GmailView, MenuControl, Model, PopupForm, PopupView, Utils, and markdownify functionality.
  * Tests cover construction, initialization, event handling, state management, integration, error handling, performance, and edge cases.
  * Enhances reliability and robustness by ensuring all major features and scenarios are thoroughly tested.

### CODE REVIEW

File: test/test_class_eventTarget.js [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+    test('should handle non-function listener in addListener', () => {
+      expect(() =>
+        eventTarget.addListener('testEvent', 'not a function')
+      ).not.toThrow();
+      expect(eventTarget._listeners.testEvent).toEqual(['not a function']);
+    });
+  });
+
+  describe('Performance Tests', () => {
+    test('should handle many listeners efficiently', () => {
```
Comment (acoven) [inline test/test_class_eventTarget.js:387]: ✅ Fixed it! Commit [b1b8280](https://github.com/appliedmedia/gmail-2-trello/commit/b1b8280)

**Review by acoven — COMMENTED** [review]
Fixed all CodeRabbit review issues

File: test/test_class_gmailView.js [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+    });
+
+    test('bindEvents should bind event listeners', () => {
+      expect(() => gmailView.bindEvents()).not.toThrow();
+    });
+
+    test('init should initialize the view', () => {
+      expect(() => gmailView.init()).not.toThrow();
+    });
+  });
```
Comment (acoven) [inline test/test_class_gmailView.js:327]: ✅ Fixed it! Commit [b1b8280](https://github.com/appliedmedia/gmail-2-trello/commit/b1b8280) - Removed tests for non-existent methods

**Review by acoven — COMMENTED** [review]
Fixed missing methods issue

File: test/test_class_menuControl.js [hunk 3]
```diff
@@ -0,0 +1,502 @@
+/**
+ * Comprehensive Jest test suite for MenuControl class
+ * Tests all methods and functionality of the MenuControl class
+ */
+
+// Mock jQuery for testing
+global.jQuery = jest.fn();
+
```
Comment (acoven) [inline test/test_class_menuControl.js:8]: ✅ Fixed it! Commit [b1b8280](https://github.com/appliedmedia/gmail-2-trello/commit/b1b8280) - Removed redundant $ mock

**Review by acoven — COMMENTED** [review]
Fixed redundant mocks

File: test/test_class_model.js [hunk 4]
```diff
@@ -0,0 +1,526 @@
...
+    test('attach should handle file upload', () => {
+      const method = 'POST';
+      const property = 'cards/123/attachments';
+      const upload1 = {
+        value: 'https://example.com/file.txt',
+        name: 'file.txt',
+      };
+      const success = jest.fn();
+      const failure = jest.fn();
+
```
Comment (acoven) [inline test/test_class_model.js:413]: ✅ Fixed it! Commit [b1b8280](https://github.com/appliedmedia/gmail-2-trello/commit/b1b8280) - Improved magic number documentation

**Review by acoven — COMMENTED** [review]
Fixed magic numbers documentation

File: test/test_class_popupView.js [hunk 5]
```diff
@@ -0,0 +1,474 @@
...
+      expect(popupView.size_k.height.max).toBeLessThanOrEqual(1400);
+    });
+
+    test('should handle window resize', () => {
+      const originalWidth = window.innerWidth;
+      const originalHeight = window.innerHeight;
+
+      window.innerWidth = 1600;
+      window.innerHeight = 900;
+
```
Comment (acoven) [inline test/test_class_popupView.js:369]: ✅ Fixed it! Commit [b1b8280](https://github.com/appliedmedia/gmail-2-trello/commit/b1b8280) - Added window state restoration

**Review by acoven — COMMENTED** [review]
Fixed window state restoration

File: test/test_class_popupView.js [hunk 6]
```diff
@@ -0,0 +1,474 @@
...
+      popupView.html = { add_to_trello: '' };
+
+      // Mock jQuery
+      const mockButton = { length: 1 };
+      const mockPopup = { length: 0 };
+      $.mockImplementation(selector => {
+        if (selector === '#g2tButton') return mockButton;
+        if (selector === '#g2tPopup') return mockPopup;
+        return { length: 0 };
+      });
```
Comment (acoven) [inline test/test_class_popupView.js:182]: 🚫 Won't do because mock setup is already in beforeEach and is test-specific

**Review by acoven — COMMENTED** [review]
Addressed mock setup issue

File: test/test_class_gmailView.js [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+      expect(() => gmailView.parseData({ force: false })).not.toThrow();
+    });
+
+    test('should handle missing DOM elements gracefully', () => {
+      gmailView.$root = null;
+      expect(() => gmailView.detectToolbar()).not.toThrow();
+    });
+  });
+
+  describe('Integration Tests', () => {
```
Comment (acoven) [inline test/test_class_gmailView.js:341]: ✅ Fixed it! Commit [b1b8280](https://github.com/appliedmedia/gmail-2-trello/commit/b1b8280) - Removed tests for non-existent methods

**Review by acoven — COMMENTED** [review]
Addressed additional missing methods

File: test/test_class_eventTarget.js [hunk 8]
```diff
@@ -0,0 +1,508 @@
...
+
+  describe('Integration Tests', () => {
+    test('should integrate with app correctly', () => {
+      expect(eventTarget.app).toBe(mockApp);
+      expect(eventTarget.app.utils).toBeDefined();
+    });
+
+    test('should work with G2T namespace', () => {
+      expect(G2T.EventTarget).toBe(EventTarget);
+    });
```
Comment (acoven) [inline test/test_class_eventTarget.js:435]: ✅ Fixed it! Added EventTarget to G2T namespace for testing.

**Review by acoven — COMMENTED** [review]
✅ Fixed it! Added EventTarget to G2T namespace for testing.

File: test/test_class_model.js [hunk 9]
```diff
@@ -0,0 +1,526 @@
...
+      expect(() => model.createCard(data)).not.toThrow();
+    });
+
+    test('uploadAttachments should upload attachments', () => {
+      const data = {
+        attachments: [{ name: 'test.txt', value: 'test-content' }],
+      };
+      expect(() => model.uploadAttachments(data)).not.toThrow();
+    });
+  });
```
Comment (acoven) [inline test/test_class_model.js:280]: ✅ Fixed it! Enhanced the test to verify actual behavior including Trello authorization check and event firing.

**Review by acoven — COMMENTED** [review]
✅ Fixed it! Enhanced the test to verify actual behavior including Trello authorization check and event firing.
